core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1502,43 +1492,62 @@ class KafkaApis(val requestChannel: RequestChannel,
       isAcknowledgeDataPresent
     }
 
+    val unauthorizedTopicPartitions: mutable.Set[TopicIdPartition] = mutable.Set.empty[TopicIdPartition]
+    val authorizedTopics = authHelper.filterByAuthorized(
+      request.context,
+      READ,
+      TOPIC,
+      shareFetchData.asScala
+    )(_._1.topicPartition.topic)
+    shareFetchData.forEach((tp, _) => {
+      if (!authorizedTopics.contains(tp.topic))
+        unauthorizedTopicPartitions.add(tp)
+    })
```

Comment:
Sorry I might didn't understand, we already have list of unauthorized topics why filtering again?

Reply:
That was really stupid, sorry about that. Made the change in. the latest commit